### ISSUE TITLE
FEATURE - [To be able to generate Employee Remuneration Report present in Xero]

### ISSUE THREAD

**Issue by aamironplan** [fivetran/dbt_xero#13]
**Are you a Fivetran customer?** 

**Aamir Sheikh, Onplan Co**

**Is your feature request related to a problem? Please describe.**
 
**Yes. We have a feature in our application in which we need to pull this Employee Remuneration Report data from Xero on our platform and allow the user to access it.**

**Describe the solution you'd like**
 
**We created a default Xero connector which is pulling complete snapshot into our bigquery database. From that database we are trying to prepare this report by doing the analysis on the extracted database snapshot.
So I wanted to take help from your support if they can assist us with a DPT package using which we can generate that Employee Remuneration Report.**

**Describe alternatives you've considered**
 
**Create a connector-cloud function and pull that data from API's. But the challenge is, its Oauth 2 based integration and we will have to keep the access token rotated to avoid expiration of it.
So is it possible we can create an oauth connector for Xero which can manage/rotate the oauth credentials and pass on the access tokens to a cloud function.
In that cloud function we will implement the logic to pull data through XERO API's using the access tokens passed by Fivetran connector.**

**Additional context**

**Please indicate the level of urgency and business impact of this request** 

**It is really critical and it is one of the crucial data we are supposed to pull from xero portal. And show to our customer in our platform. It would really affect our relationship with the customer. It might result penalty as well if we will not deliver it in time. **

**Are you interested in contributing to this package?** 

No, I'd prefer if someone else did this.  I don't have the time and/or don't know how to incorporate the changes necessary.

**Comment by fivetran-joemarkiewicz** [comment]
Hi @aamironplan thanks so much for opening the feature request for Xero! 

I can definitley see this final model for the Employee Remuneration being a good addition. Would you be able to confirm that [this](https://central.xero.com/s/article/Employee-Remuneration-Details-report-AU) is the report you are referencing from Xero?

If so, I actually think you may be able to recreate this report for the time being outside of the dbt package using the below highlight raw tables that should be included in your base Xero connector.
![image](https://user-images.githubusercontent.com/74217849/134236280-fa73a295-07f1-454b-be66-b23a405762b1.png)

Let me know if you are able to take a look at the base tables and if these could possibly get you to where you need to be with the Employee Remuneration report.

Thanks!

**Comment by aamironplan** [comment]
Yes this is the report : https://central.xero.com/s/article/Employee-Remuneration-Details-report-AU
I tried but having difficulties in connecting the dots. I am looking for an expert from your team who has good amount of knowledge of its database if any who can help us in understanding.
OR if you can provide a DPT package then it will be really great.

**Comment by fivetran-joemarkiewicz** [comment]
Our team would be happy to meet with you during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours?month=2021-09) to try and help to connect the dots and chat more about this if you would like.
